scapy/fields.py

```diff
@@ -457,11 +516,14 @@ def do_copy(self, x):
         else:
             return [p if isinstance(p, six.string_types) else p.copy() for p in x]
     def getfield(self, pkt, s):
-        c = l = None
+        c = l = cls = None
         if self.length_from is not None:
             l = self.length_from(pkt)
         elif self.count_from is not None:
             c = self.count_from(pkt)
+        if self.next_cls_cb is not None:
```

Comment:
You may want to use `elif` here.

Reply:
`elif` is not appropriate here, because `length_from` and `next_cls_cb` can be used concurrently. Using `elif` would prevent that. I could rewrite it like this, though:
```
if self.count_from(pkt) is not None:
    c = self.count_from(pkt)
else:
    if self.length_from is not None:
        l = self.length_from(pkt)
    if self.next_cls_cb is not None:
        cls = self.next_cls_cb(pkt, [], None, s)
        c = 1
```

Would you prefer this approach? 

It is worth noting that this breaks compatibility with existing scripts that are specifying both `count_from` and `length_from` (although this is meaningless). Indeed, with the code in the master branch, `length_from` would be used and `count_from` would be ignored. With this proposal, `count_from` would be used and `length_from` would be ignored.